<launch>

<!-- Launch Epoch Clock without params (=realtime, start from now) -->
    <node pkg="epoch_clock" name="epoch_clock_sim" type="epoch_clock.py" output="screen">
        <param name="realtime_factor" value="1" />
    </node>

    <group ns="cso">
        <!-- Load Configuration File for sensors etc -->
        <rosparam command="load" file="$(find cso_gnc_sim)/cfg/cso.yaml" />

        <!-- Launch SGP4 node with TLE (cheating here and using swisscube TLE) -->
        <node pkg="sgp4_node" name="sgp4_node_cso" type="sgp4_node.py" output="screen">
            <param name="tle_line1" value="1 35932U 09051B   17242.83495672  .00000195  00000-0  55221-4 0  9991" />
            <param name="tle_line2" value="2 35932  98.5062  22.7182 0008099  30.5945 329.5723 14.56041441421185" />
        </node>

        <!-- Launch TF publisher for visualizations -->
        <node pkg="cso_gnc_sim" name="tf_pub" type="satelite_tf_publisher.py" output="screen">
            <remap from="pose" to="/cso/pose" />
            <remap from="~sensors" to="/cso/sensors" />
            <param name="body_frame" value="cso" />
            <param name="no_position" value="0" />
        </node>

        <!-- Launch Radar Sensor -->
        <node pkg="space_sim_radar" name="node_radar1" type="sim_radar_node.py" output="screen">
            <remap from="~sensor" to="/cso/sensors/Radar" />
        </node>
    </group>

</launch>
